### DISCUSSION TITLE
Suggestions for Project Enhancement

### DISCUSSION THREAD

**Discussion by gg64nou** [Part-DB/Part-DB-server#320] [Ideas]
Hello, first of all, thank you very much for this wonderful project.

I have a few suggestions :D

1. If I have 5 parts in location X and I want to move them all to location Y (which has 0 parts), currently I have to go to "Edit" mode, but I think this can also be done through the "Move parts from lot to another lot" button. This is what I need the most. If the location I want to move a part to is not declared, I can't move it even if the location exists. I have to go to "Edit" mode and change it from there.

2. For the parts marked as "Needs refill," it would be nice to have a page where they can all be viewed (this can also be done from "All parts").

3. A special page with all the parts that I have ordered and are on their way. This way, I know that I have already ordered those parts and won't place another order for the same parts. When they arrive, I can move them to a location.

4. Borrowing parts: If I lend a part to a friend, I can click the minus button to indicate that, but I don't have a specific list for it. Here's a little trick I came up with: I create a location where I move the parts that I lend.

**Comment by jbtronics** [comment]
> 2\. For the parts marked as "Needs refill," it would be nice to have a page where they can all be viewed (this can also be done from "All parts").

You can already do this in the filter panel by setting the "Any lot needs refill" checkbox in the "Stocks" tab.



> 3\. A special page with all the parts that I have ordered and are on their way. This way, I know that I have already ordered those parts and won't place another order for the same parts. When they arrive, I can move them to a location.

Currently, there is no real possiblity to mark parts as ordered. A system for managing orders is planned for the future, but that will take some time. Currently, I am working on a system which allows Part-DB to utilize distributor APIs to automatically retrieve part information (like paramaters, datasheets and more)

**Reply by gg64nou** [reply]
Thank you for the response. :D

> You can already do this in the filter panel by setting the "Any lot needs refill" checkbox in the "Stocks" tab.

Yes, I know, I was referring to a quick way to view that filter, maybe a setting to input a few custom links with the filters we would need quickly.

> Currently, there is no real possiblity to mark parts as ordered. A system for managing orders is planned for the future, but that will take some time. Currently, I am working on a system which allows Part-DB to utilize distributor APIs to automatically retrieve part information (like paramaters, datasheets and more)

+1 for the API. Sounds great, looking forward to it ^.^. 
Thank you once again for everything you do
